### PR TITLE
add enforce condition to return unauthorized

### PR DESCRIPTION
**Pull request by acosferreira**
# Description
Include validation to return 501 when not enforced edge groups

FIXES: <THEEDGE-3815>

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

**Review by omaciel — COMMENTED** [review]
Hmmm why is go.mod updated?

**Comment by acosferreira** [comment]
@omaciel they complaint about no required module provides package github.com/swaggo/swag; to add it:
go get github.com/swaggo/swag

File: pkg/routes/devicegroups.go [hunk 1]
```diff
@@ -323,7 +323,7 @@ func GetAllDeviceGroups(w http.ResponseWriter, r *http.Request) {
 func CreateDeviceGroup(w http.ResponseWriter, r *http.Request) {
 	ctxServices := dependencies.ServicesFromContext(r.Context())
 
-	if feature.HideCreateGroup.IsEnabled() {
+	if feature.HideCreateGroup.IsEnabled() || !(feature.EnforceEdgeGroups.IsEnabled()) {
```
Comment (ldjebran) [inline pkg/routes/devicegroups.go:326]: 1. The task is disable the feature when inventory groups feature is enabled and not a special org , this should be applyed for create update delete and add systems. 
2. prefer feture not available returned like here https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/thirdpartyrepo.go#L127

```go
if feature.EdgeParityInventoryGroupsEnabled.IsEnabled() && !utility.EnforceEdgeGroups(orgID) {
    respondWithAPIError(w, ctxServices.Log, errors.NewFeatureNotAvailable(""))
    return 

}
```

**Review by ldjebran — COMMENTED** [review]
left a comment, all should be covered by unit-tests.

**Comment by ldjebran** [comment]
@acosferreira for dependencies , 

run:  go install -a 

and run: go mod tidy

**Pull request closed without merging**